### PR TITLE
#4562: normalize global operator

### PR DESCRIPTION
**Pull request by muthutt**
normalize global operator : compute global norm for a given rank-4 tensor

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/composite/composite_ops.cpp [hunk 1]
```diff
@@ -938,6 +938,35 @@ Tensor normalize_hw(const Tensor& y,const MemoryConfig& output_mem_config) {
...
 
+
+using HWFunctionT = std::function<Tensor (const Tensor& y, const MemoryConfig&)>;
+Tensor _make_global_from_hw_impl(HWFunctionT fn, const Tensor& y, const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG) {
+    const Shape s_orig = y.shape();
+    TT_FATAL(s_orig.rank() == 4, "Cannot support non-rank 4 Tensor");
+
+    //format to HW
+    Tensor y_chw = reshape(y,1,s_orig[1],s_orig[2],-1);
+    Tensor y_hw = reshape(y_chw,1,1,s_orig[2],-1);
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/composite/composite_ops.cpp:949]: Could you explain why two reshapes are needed here, and similarly for reformatting the output?

**Review by eyonland — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/composite/composite_ops.cpp (hunk 1 above)
Comment (tt-aho) [reply tt_eager/tt_dnn/op_library/composite/composite_ops.cpp:949]: What are the limitations/issues with running reshape only once instead of twice?

**Comment by tt-aho** [comment]
Reshape should work on multiple dims. Can you retry and post if there is an issue?

**Comment by tt-aho** [comment]
> > Reshape should work on multiple dims. Can you retry and post if there is an issue?
> 
> when was this change made ? curious

I don't recall there ever being an issue with reshape only working on one dim at a time.

File: tt_eager/tt_dnn/op_library/composite/composite_ops.cpp [hunk 2]
```diff
@@ -938,6 +938,32 @@ Tensor normalize_hw(const Tensor& y,const MemoryConfig& output_mem_config) {
...
+    TT_FATAL(s_orig.rank() == 4, "Cannot support non-rank 4 Tensor");
+
+    //format to HW
+    Tensor y_hw = reshape(y,1,1,s_orig[2],s_orig[3]*s_orig[1]*s_orig[0]);
+
+    // compute @fn
+    Tensor z_0 = fn(y_hw,output_mem_config);
+    TT_FATAL( y_hw.shape() == z_0.shape(), "shape match" );
+
+    // reformat
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/composite/composite_ops.cpp:953]: You should be able to deallocate y_hw at this point

**Review by tt-aho — APPROVED** [review]

**Pull request merged**